File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/highriskpostcodesupload/PostDistrictIndicators.java
package uk.nhs.nhsx.highriskpostcodesupload;

import com.fasterxml.jackson.annotation.JsonCreator;

public class PostDistrictIndicators {

    public final String riskIndicator;
    public final String tierIndicator;

    @JsonCreator
    public PostDistrictIndicators(String riskIndicator, String tierIndicator) {
        this.riskIndicator = riskIndicator;
        this.tierIndicator = tierIndicator;
    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/highriskpostcodesupload/PostDistrictIndicators.kt
package uk.nhs.nhsx.highriskpostcodesupload

import dev.forkhandles.values.NonBlankStringValueFactory
import dev.forkhandles.values.StringValue

enum class RiskIndicator(val wireValue: String) {
    HIGH("H"), MEDIUM("M"), LOW("L");

    override fun toString() = wireValue

    companion object {
        fun from(wire: String) = values().first { it.wireValue == wire }
    }
}

class TierIndicator private constructor(value: String) : StringValue(value) {
    companion object : NonBlankStringValueFactory<TierIndicator>(::TierIndicator)
}

data class PostDistrictIndicators(
    val riskIndicator: RiskIndicator,
    val tierIndicator: TierIndicator
)
